Travel Scheduler/Views/SlideMenuUIView.m

```diff
@@ -0,0 +1,90 @@
+//
+//  SlideMenuUIView.m
+//  Travel Scheduler
+//
+//  Created by frankboamps on 7/22/19.
+//  Copyright © 2019 aliu18. All rights reserved.
+//
+
+#import "SlideMenuUIView.h"
+#import "SlideMenuTableViewCell.h"
+
+@interface SlideMenuUIView()<UITableViewDataSource, UITableViewDelegate>
+
+@end
+
+@implementation SlideMenuUIView
+
+#pragma mark - Setting view implementation
+
+- (void)loadView
+{
+    menuArray =[NSArray arrayWithObjects:@"Profile",@"Friends",@"Status",@"Settings",@"Logout",nil];
+    self.slideInTableView = [[UITableView alloc] initWithFrame:CGRectMake(CGRectGetWidth(self.frame), 0, 0 , 4000)];
+    self.slideInTableView.backgroundColor = [UIColor whiteColor];
+    [self.slideInTableView setAutoresizesSubviews:YES];
+    [self.slideInTableView setAutoresizingMask:(UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight)];
+    self.slideInTableView.delegate = self;
+    self.slideInTableView.dataSource = self;
+    [self addSubview: self.slideInTableView];
+    [self addSubview: self.closeSlideInTableViewButton];
+}
+
+#pragma mark - Button for slide in view
+
+- (void)createButtonToCloseSlideIn
+{
+    self.closeSlideInTableViewButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    self.closeSlideInTableViewButton.backgroundColor = [UIColor whiteColor];
+    [self.closeSlideInTableViewButton setFrame:CGRectMake(CGRectGetWidth(self.frame)-30, 100, 20 , 20)];
+    [self.closeSlideInTableViewButton setBackgroundImage:[UIImage imageNamed:@"close_icon"] forState: UIControlStateNormal];
+    self.closeSlideInTableViewButton.layer.cornerRadius = 10;
+    self.closeSlideInTableViewButton.clipsToBounds = YES;
+    [self.closeSlideInTableViewButton addTarget: self action: @selector(buttonClicked:) forControlEvents: UIControlEventTouchUpInside];
+    [self addSubview: self.closeSlideInTableViewButton];
+}
+
+- (void)buttonClicked:(id)sender
+{
+    [self animateViewBackwards];
+}
+
+#pragma mark - Animations for slide out view
+
+- (void) animateViewBackwards
+{
+    [UIView animateWithDuration: 0.50 animations:^{
+        self.frame = CGRectMake(CGRectGetMaxX(self.frame), 0, 300 , 4000);
+    }];
+}
+
+#pragma mark - TableView DataSource methods
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    static NSString *cellIdentifier = @"SlideMenuTableViewCell";
```

Comment:
Aren't static variables usually declared outside of the implementation?

Reply:
Yes, but it's not a problem to do this either...